container/network.go

```diff
@@ -34,7 +43,10 @@ func (c *Container) CreateNetwork(namespace []string) (id string, err error) {
 }
 
 // DeleteNetwork deletes a Docker Network associated with a namespace.
-func (c *Container) DeleteNetwork(namespace []string) error {
+// event parameter can be "destroy" or "remove". If the network was used by a service, the event to use is "destroy". If the network has not been used, the event is "remove".
+// Remove removes the reference from Docker to the network.
+// Destroy removes the network from Docker active network.
+func (c *Container) DeleteNetwork(namespace []string, event EventType) error {
```

Comment:
it's a bit odd to me having _event_ as an argument to this func because it does not change the behavior of network removing. we can listen for both events and return after the first received event. this way use of this func will be agnostic and caller will not need to figure out which event type it should pass.

Reply:
The problem with this one is that we cannot wait for both. In some cases we will have both (when we create the service and attach the network to this service) but in some other cases when we just create and destroy the network without any service we only have the "remove" event.

I agree this is kind of weird to have this `event` maybe we can rename it to `deletionEvent` or something like that.

I don't see any ways to make this function agnostic because it's really related to docker and how we use it but what thing that can be done in the future is to remove this `CreateNetwork` and `DeleteNetwork` from the public API and manage the network internally in the container package which totally makes sense.

Reply:
Totally agree with the last part.
> This is not a perfect solution but the perfect solution requires to reorganize the container package to match our logic (mesg's service have multiple docker's service, one network) rather than staying close to the docker logic (service, container, network). By doing this, we could extract a lot of docker related logic from the service package to the container package (like start and stop mesg's service with dependencies and network).

Reply:
i don't quite get the limitation of waiting for both _destory_ and _remove_ here. does docker produces two events for both _destory_ and _remove_ when we have a service attached to network?

Reply:
Yes it does. But as we run integration test that only create a network and them delete it, we need a way to change the behaviour of `DeleteNetwork` otherwise we have timeout and the integration test on `CreateNetwork` doesn't work (as well as others).

Reply:
That's why, after transferring more logic to the container package (where we ALWAYS create a network attached to service), we will be able to remove the event parameter.

Reply:
I understand now. maybe we should inspect the network to see if there is any services attached to it before removing, this way we can guess the event type fore network removing without doing changes on container package. but there can be a race condition with this approach, while we're inspecting the network some services can be removed or started.

Reply:
i think, the race condition can be prevented by also listening for service start/stop events in this func.